request-server_test.go

```diff
@@ -714,9 +754,17 @@ func TestRequestReadlink(t *testing.T) {
 	require.NoError(t, err)
 	err = p.cli.Symlink("/foo", "/bar")
 	require.NoError(t, err)
+
 	rl, err := p.cli.ReadLink("/bar")
 	assert.NoError(t, err)
-	assert.Equal(t, "foo", rl)
+	assert.Equal(t, "/foo", rl)
+
+	_, err = p.cli.ReadLink("/foo")
+	assert.Error(t, err, "Readlink on non-symlink should fail")
+
+	_, err = p.cli.ReadLink("/foobar")
```

Comment:
I recommend using a filename that within-itself says it should not exist when testing something on a file that should not exist. I might accidentally make a `/foobar` file/directory and not realize it would break another test. But I’m far less likely to accidentally make a file `/does-not-exist`.

Reply:
@georgmu you missed this one. I'll do this change after merging, thanks!

Reply:
> Sorry, I had it in a commit here, but made a mistake while rebasing and resorting of commits...

No problem, thanks for your work